tests/testapp/test_models.py

```diff
@@ -236,6 +237,22 @@ def test_use_index_primary(self):
         used = used_indexes(cap.query)
         assert len(used) == 0 or "PRIMARY" in used
 
+    def test_use_index_inner_query(self):
+        title_idx = index_name(Book, "title")
+        with CaptureLastQuery() as cap:
+            list(
+                Author.objects.annotate(
+                    has_books=Exists(
+                        Book.objects.filter(author_id=OuterRef("id"), title__gt="")
+                    )
+                )
+                .filter(has_books=True)
+                .use_index(title_idx, table_name="testapp_book")
+            )
+        assert ("USE INDEX (`" + title_idx + "`)") in cap.query
+        used = used_indexes(cap.query)
+        assert len(used) == 0 or title_idx in used
```

Comment:
Why is there an `OR`? I'd prefer if tests were deterministic

Reply:
It's the same on the two tests above. the assert on line 252 is deterministic, but whether mySQL actually decides to use it in the query plan despite the USE index hint, which is the result of used_indexes, it just seems to run .explain() is not something that can be controlled from the ORM side.

Reply:
Ah yes - I haven't worked on this in years :)